Add support for I2C on STM32F7 family and on STM32F723E-DISCO board

aurel32 opened the pull request:
This patch series adds support for I2C bus of the STM32F7 family, enables it on the STM32F723E-DISCO board, and finally fixes an issue found while testing the I2C support in interrupt mode.

I have tested it using by communicating with the WM8994 chip on the STM32F723E-DISCO board.

aurel32 committed:
boards: arm: stm32f723e_disco: enable I2C support

Enable the support for the 3 I2C buses of the STM32F723E-DISCO board,
connected respectively to the WM8994, the Arduino V3/STMOD+ and the
touch panel.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>

ydamigos reviewed (changes requested):
Looks good. I am just blocking it because PR #7626 also introduces the NACK fix.

@aurel32 could you coordinate with @vaussard on this?

aurel32 commented:
@ydamigos, thanks for the pointer. Given there is already a PR opened about that issue, I believe it's better to discuss it there. In addition the fix is now conflicting with the changes introduced by PR#5224, and it more work is needed to understand how to handle a NACK in slave mode. I'll therefore just drop that commit from the pull request.

aurel32 committed:
drivers: i2c: stm32: add support for STM32F7

The STM32F7 uses the V2 version of the STM32 I2C controller. Add the
corresponding Kconfig, DTS, DTS fixup and pinmux entries.

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>

aurel32 committed:
drivers: i2c: stm32: Kconfig: Remove redundant 'default n' properties

Bool symbols implicitly default to 'n'.

A 'default n' can make sense e.g. in a Kconfig.defconfig file, if you
want to override a 'default y' on the base definition of the symbol. It
isn't used like that on any of these symbols though, and is
inconsistent.

This will make the auto-generated Kconfig documentation have "No
defaults. Implicitly defaults to n." as well, which is clearer than
'default n if ...'

This is basically reapplying commit 133a299b50 ("drivers: i2c: Kconfig:
Remove redundant 'default n' properties"), which has been partially
reverted in commit c7875b75aa ("i2c: stm32_v2: implement slave
support").

Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>

aurel32 commented:
And I have just added an additional cleanup commit to make @ulfalizer happy.

aurel32 commented:
@ydamigos, there was a third PR with a fix for this issue, PR #5224. Given it has just been merged, i confirm that just removing the commit from my PR was the right thing to do.

erwango commented on boards/arm/stm32f723e_disco/stm32f723e_disco_defconfig:
```diff
@@ -12,6 +12,9 @@ CONFIG_SERIAL=y
 CONFIG_UART_CONSOLE=y
 
+# enable I2C
+CONFIG_I2C=y
```
Should be 'n' by default.

ydamigos replied:
@erwango We enable buses on board level when there is a peripheral (e.g. WM8994 or touch panel in this case) that needs them, aren't we?

erwango replied:
@ydamigos : we should configure them, and this is what is done here. But activation should always be done in application and not default.
Aim is that several developers on the same board know what is the base components for the board so they only enable what they need. But they don't have to bother disabling what other developers may have enabled meanwhile.

ydamigos replied:
@erwango We need to clean up board _defconfig files then.

erwango replied:
@ydamigos : yes, there is some work ahead! I raised https://github.com/zephyrproject-rtos/zephyr/issues/7151 earlier, waiting https://github.com/zephyrproject-rtos/zephyr/pull/7176 to be reviewed and used a reference (which was locked waiting for new SDK version and which I hope should get some feedbacks soon)

erwango reviewed (changes requested):
One comment.

Otherwise, could you update help message for I2C_STM32_V2. Message is not quite clear but we should be explicit that drivers now supports F7 (while previously it was 'compatible': I agree this is subtle :-))

aurel32 commented:
@erwango: I have just done the requested changes, i.e. I updated the `Kconfig.stm32` help message to mark F7 as supported, and removed `CONFIG_I2C=y` from the board defconfig.

aurel32 commented:
@erwango, I have just pushed a new version to solve the conflict introduced by PR #8663 and to add the new clock properties introduced in PR #8759.

dbkinder reviewed (approved):
Doc changes LGTM, thanks.

aurel32 commented:
I have just pushed a new version to solve the conflict due the merge of USB OTG FS support. There is no code change beside solving conflicts.

aurel32 commented:
I have just pushed a new version to solve the conflict due the merge of USB OTG HS support. There is no code change beside solving conflicts.
